consensus/consensus.go

```diff
@@ -98,4 +99,13 @@ type PoW interface {
 
 	// Hashrate returns the current mining hashrate of a PoW consensus engine.
 	Hashrate() float64
+
+	// CalcDifficulty returns the difficulty that a new block should have when created at time
+	// given the parent block's time and difficulty.
+	CalcDifficulty(*params.ChainConfig, uint64, *types.Header) *big.Int
```

Comment:
Could we perhaps similarly use `consensus.Engine.Prepare` to calculate the difficulty? Maybe some code needs the raw difficulty, just asking. Ether way, it should probably land in the Engine interface, not in PoW only.